Evaluation-2/src/expression-test.js

```diff
@@ -1,127 +1,92 @@
-import { createStack } from "./stack.js"
-export function expressionTest(){
-let stack=createStack()
-return{
-    checkExpr(expr){
-        let str=expr;
-        let expArr=[]
-        for(let i=0;i<str.length;i++){
-            if(str[i]=='1'||str[i]=='2'||str[i]=='3'||str[i]=='4'||str[i]=='5'||str[i]=='6'||str[i]=='7'||str[i]=='8'||str[i]=='9'||str[i]=='+'||str[i]=='-'){
-
-                expArr.push(str[i])
-            }
+/**
+ * function takes an array containing expression 
+ * @param {*} arr computes the array and returns it
+ * @returns a computed or evaluated expression
+ */
+function compute(arr){
+    let operandSt=[]
+    let operatorSt=[]
+    let result=[]
+    for(let a of arr){
+        if(a=='0'||a=='1'||a=='2'||a=='3'||a=='4'||a=='5'||a=='6'||a=='7'||a=='8'||a=='9'){
```

Comment:
This can be simplified actually. Dont need to check each digit like this. First convert to a number and check if it is in the range 0-9

Reply:
fixed it.